ouroboros-consensus/src/Ouroboros/Storage/VolatileDB/Types.hs

```diff
@@ -31,11 +31,17 @@ type Index blockId = Map String (Maybe Slot, Int, Map Int64 (Int, blockId))
 -- in bytes.
 type ReverseIndex blockId = Map blockId (String, Int64, Int)
 
+-- Controls how the db behaves if it finds files with less blocks than expected.
+data BlocksPerFileMode =
+      Strict -- Will raise an exception, unless only the latest file has less blocks.
+    | LenientFillOldFirst -- Here anexception will not be raised and small files
+                          -- will be written first.
+    | LenientFillOnlyLast -- Only the last file can be written.
```

Comment:
I think we don't need this option; let's hardcode the behaviour of the final constructor.

Reply:
Let's add a comment to `reopen` stating the invariant that closing and reopening the DB should be a no-op (in particular, will continue to write to the same file).